R/request_figma.R

```diff
@@ -0,0 +1,84 @@
+#' Create a request for a specified Figma API Endpoint
+#'
+#' @rdname request_figma_endpoint
+#' @inheritParams httr2::request
+#' @inheritParams httr2::req_method
+#' @inheritParams req_figma_template
+#' @inheritParams req_rigma_agent
+#' @inheritDotParams httr2::req_template
+#' @keywords internal
+#' @importFrom httr2 request req_template req_error resp_body_json req_method
+#' @importFrom rlang is_null
+request_figma_endpoint <- function(endpoint = NULL,
+                                   base_url = "https://api.figma.com",
+                                   template = NULL,
+                                   method = NULL,
+                                   token = NULL,
+                                   ...) {
+  req <- httr2::request(base_url) %>%
+    req_figma_template(
+      endpoint = endpoint,
+      template = template,
+      ...
+    ) %>%
```

Comment:
This is template is making things complex. Maybe we should not introduce.
The approach looks more generic that it needs to be. 
It shifts the usage of `req_url_path_append()` to pasting strings with `glue()` and introduces an unnecessary ellipses `...`.

Reply:
Hmmm. Under the hood, `httr2::req_template()` is using `paste0()` not glue and the ellipses are how the parameters for each template are passed. I actually really like this pattern since it allows you to see all of the API URL patterns in one place, potentially consolidate parameter checks, and avoid duplication within the public facing functions. I could merge `req_figma_template()` back into `request_figma()` but I think there was at least one function that required the flexibility of two separate functions.

Is your concern with performance? Or maintainability? Happy to explore alternatives if I can get a better understanding of the issue.

Reply:
Concern is maintenance. Let's continue the discussion here: https://github.com/AleKoure/Rigma/pull/9#discussion_r1234779897